collections/join_to_string.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+/**
+ * Options for joinToString
+ */
+export type JoinToStringOptions = {
+  separator?: string;
+  prefix?: string;
+  suffix?: string;
+};
+
+/**
+ * Transforms the elements in the given array to strings using the given selector.
+ * Joins the produced strings into one using the given separator and applying the given prefix and suffix to the whole string afterwards.
+ * Returns the resulting string.
+ *
+ * Example:
+ *
+ * ```ts
+ * import { joinToString } from "./join_to_string.ts";
+ * import { assertEquals } from "../testing/asserts.ts";
+ *
+ * const users = [
+ *     { name: 'Kim' },
+ *     { name: 'Anna' },
+ *     { name: 'Tim' },
+ * ];
+ *
+ * const message = joinToString(users,
+ *     (it) => it.name,
+ *     {
+ *       separator: " and ",
+ *       prefix: "<",
+ *       suffix: ">",
+ *     },
+ * );
+ *
+ * assertEquals(message, "<Kim and Anna and Tim>");
+ * ```
+ */
+export function joinToString<T>(
+  array: readonly T[],
+  selector: (el: T) => string,
+  {
+    separator = ", ",
```

Comment:
I think space separator is more appropriate

```suggestion
separator = " ",
```

Reply:
`Array.join` and Kotlin's `joinToString` both have the comma in their default, why should we deviate?

Reply:
@zhangyuannie @LionC `Array.prototype.join` default separator is `","` but Kotlin's `joinToString` default separator is `", "`. 

I copied my default parameters from Kotlin's `joinToString` but I guess it's better to go with `Array.prototype.join` in this case. more familiar API for JS developers.